sunpycube/cube/NDCube.py

```diff
@@ -241,7 +241,17 @@ def __getitem__(self, item):
             mask = self.mask[item]
         else:
             mask = None
-        result = NDCube(data, wcs=wcs, mask=mask, uncertainty=self.uncertainty,
+        if self.uncertainty is not None:
+            if not isinstance(self.uncertainty.array, int):
```

Comment:
I think assuming it is an int is too specific.  It's more likely to be a float.  Perhaps you should turn the if statement around and check that if self.uncertainty IS array like.